addons/account_payment/tests/test_account_payment.py

```diff
@@ -185,3 +185,32 @@ def test_prevent_unlink_apml_with_active_provider(self):
         self.assertEqual(self.dummy_provider.state, 'test')
         with self.assertRaises(UserError):
             self.dummy_provider.journal_id.inbound_payment_method_line_ids.unlink()
+
+    def test_duplicate_provider_child_company_no_journal_id(self):
```

Comment:
This file is only for testing the `account.payment` model; please move the test to a new `test_payment_provider.py` file.